app/Compile.hs

```diff
@@ -3,6 +3,7 @@
 module Compile where
```

Comment:
Wonder whether this module should live in src. It would be great to have it backend agnostic. What do you think of having a Backend typeclass?

Reply:
Yes, this definitely has to go and should move to, I guess, `src/`. Having a Backend typeclass is what I was thinking about as well, but I need to think it through with regards to how this will work together with the  Juvix-module level backends.

Reply:
For the Backend class, see: https://github.com/heliaxdev/juvix/issues/157#issuecomment-793964127